lib/rewards/client/rewards-message-listener.js

```diff
@@ -0,0 +1,116 @@
+const { sc } = require('@reldens/utils');
+const { RewardsConst } = require('../constants');
+const { GameConst } = require('../../game/constants');
+
+class RewardsMessageListener
+{
+
+    static listenMessages(room, gameManager)
+    {
+        room.onMessage('*', (message) => {
+            let rewards = sc.get(message, RewardsConst.REWARDS_CONST, false);
+            if (rewards) {
+                this.loadRewards(rewards, gameManager);
+            }
+
+            if (RewardsConst.REMOVE_DROP === message.act) {
+                this.removeRewardById(message.id, gameManager);
+            }
+        });
+    }
+
+    static loadRewards(rewards, gameManager)
+    {
+        let currentScene = gameManager.getActiveScene();
+        let gameConfig = gameManager.config;
+        let objectPlugin = gameManager.getFeature('objects');
+        let loader = currentScene.load;
+        // TODO: Validate previous required gameManager sections exists.
+
+        for (let [rewardId, reward] of Object.entries(rewards)) {
+            reward['params'] = reward[RewardsConst.REWARDS_PARAMS];
+            this.loadSpritesheet(reward, loader, gameConfig);
+            loader.once('complete', async () => {
+                await this.createRewardAnimation(objectPlugin, reward, rewardId, currentScene);
+            });
+        }
+        loader.start();
+    }
+
+    static sendRewardPickUp(rewardId, activeRoomEvent)
+    {
+        let dataSend = {
+            act: RewardsConst.REWARDS_PICK_UP_ACT,
+            id: rewardId
+        };
+        activeRoomEvent.room.send('*', dataSend);
+    }
+
+    static async createRewardAnimation(objectsPlugin, reward, rewardId, currentScene)
+    {
+        return await objectsPlugin.createAnimationFromAnimData({
+            type: RewardsConst.REWARDS_PICK_UP_ACT,
+            enabled: true,
+            ui: true,
+            frameStart: reward.params['start'],
+            frameEnd: reward.params['end'],
+            repeat: -1,
+            autoStart: true,
+            key: rewardId,
+            id: 100,
```

Comment:
hardcoded ID: 100?

Reply:
probably testing what that id parameter was for. the odd thing is that everything was working fine with that hardcoded. Gonna change it for `rewardId`